### ISSUE TITLE
Driver autocomplete flow card event handler is not passed the associated device

### ISSUE THREAD

**Issue by netactivenl** [athombv/homey#234]
**What is your Homey version (Settings → System)?**

0.8.17

**What did you try to achieve?**

I want to use an autocomplete argument on a flow card to offer the user a selection of device activities to choose from.

**What was the result?**

In my autocomplete event handler I only get the first letter I typed in the autocomplete field. Subsequent letters I type do raise the event again, but keep getting only the first letter. 

**What did you expect as result?**

I expected to see events come in as I type, everytime giving passing the complete current text in the argument field. Also I was expecting to get the device associated with the card, but the device is never passed. Emile stated I should be getting something like this:

{ device: ​*device_data*​, query: “start tv” }

But I'm only getting:

{ query: '' } or { query: 's' }

This is my app.json (relevant part only):

`"flow": {
    "actions": [
      {
        "id": "start_activity",
        "title": {
          "en": "Start an activity",
          "nl": "Start een activiteit"
        },
        "args": [
          {
            "name": "hub",
            "type": "device",
            "filter": "driver_id=hub"
          },
          {
            "name": "activity",
            "type": "autocomplete",
            "placeholder": {
              "en": "Activity name",
              "nl": "Naam activiteit"
            }
          }
        ]
      }
    ]
  }`

**Comment by matjaz** [comment]
something is inconsistent [in docs](https://developers.athom.com/library/flow)
`Homey.manager('flow').on('action.card_id.arg_name.autocomplete', function( callback, value ){`
what should the `value` be?
In [Youtube app](https://github.com/matjaz/homey-youtube) I get `{query: 'value from input'}`

**Comment by netactivenl** [comment]
You are correct that the docs are inconsistent. Value should contain both the flow card's device (if any) and the value entered by the user in the following format (where "start tv" would be the text entered by the user in the autocomplete field):

{ device: ​device_data​, query: “start tv” }

**Comment by matjaz** [comment]
Maybe as a feature request, other arguments could be also sent.

**Comment by PatrickSannes** [comment]
In my code all auto complete works like it should. I have an array with an filter:

```
    Homey.manager('flow').on('action.increment_number_variable.variable.autocomplete', function (callback, value) {
        callback(null, variableManager.getVariables().filter(util.filterVariable(value, 'number')));
    });
exports.filterVariable = function(partialWord, type) {
    return function(element) {
        return element.name.toLowerCase().indexOf(partialWord.query.toLowerCase()) > -1 && element.type === type;
    }
}

```

**Comment by netactivenl** [comment]
Maybe it has to do with the fact my flow card is part of a driver. Is your flow card part of an app you drag from the App container in the sidebar?

**Comment by PatrickSannes** [comment]
That could be. Mine is part of an app.

**Comment by WeeJeWel** [comment]
Updated the docs. Please check if the issue still persists in 0.8.19 when it's released.

**Comment by netactivenl** [comment]
I'll check it once I get the update. For now are the updated docs published already? I don't read anything about the device that should also be passed, @WeeJeWel

**Comment by netactivenl** [comment]
@WeeJeWel This issue still exists in 0.8.20 (experimental). An autocomplete event handler doesn't get the device passed that is associated with the card.

**Comment by WeeJeWel** [comment]
Fixed in 0.8.21

**Comment by netactivenl** [comment]
This issue is fixed in 0.8.22. Tnx @WeeJeWel !
